packages/smooth_app/lib/widgets/smooth_product_carousel.dart

```diff
@@ -279,17 +280,26 @@ class _SearchCardTagLine extends StatelessWidget {
               future: _fetchData(),
               builder: (BuildContext context,
                   AsyncSnapshot<Map<String, dynamic>> data) {
-                if (data.data == null || !data.hasData) {
+                if (data.connectionState == ConnectionState.done) {
+                  if (data.hasData) {
+                    if (data.data![TAG_LINE_KEY] != null) {
+                      tagline = data.data![TAG_LINE_KEY] as TagLineItem;
+                      return _SearchCardTagLineText(
+                        tagLine: tagline!,
+                      );
+                    }
+                    if (data.data![DEPRECATED_KEY] as bool) {
+                      return const _SearchCardTagLineDeprecatedAppText();
+                    }
+                  }
                   return const _SearchCardTagLineDefaultText();
                 }
 
-                if (data.data![DEPRECATED_KEY] as bool) {
-                  return const _SearchCardTagLineDeprecatedAppText();
-                }
-
-                if (data.data![TAG_LINE_KEY] != null) {
+                ///This is to avoid blinking issue caused, Due to [ConnectionState.done] followed by [ConnectionState.waiting]
+                ///When [Consumer] refreshes the [FutureBuilder] causing [_fetchData()] to be called again.
+                if (tagline != null) {
```

Comment:
I have good news: you probably don't need the `Consumer` at all, as the only added value (?) it provides is setting a default value in one specific case. And the default value in the other cases is also the same value.

Reply:
> I have good news: you probably don't need the `Consumer` at all, as the only added value (?) it provides is setting a default value in one specific case. And the default value in the other cases is also the same value.

@monsieurtanuki  I wasn't able to see the tagline untill, I did my first scan.I guess we will need `UserPreferences` instance to check the same.

Reply:
@omkarChend1kar Actually you're right (I won't discuss here the functional added value of waiting for the first scan).
That said, we may agree that

Reply:
Oops...

@omkarChend1kar Looking again at the code, this is what I would suggest:
* replace the `Consumer<UserPreferences>` syntax with `context.watch<UserPreferences>()` - not because it's smarter but because it is consistent with the rest of the app, for maintenance purposes
* run the `fetchData` only once by storing the `Future` in a variable, like we do in `main.dart`
```dart
  // We store the argument of FutureBuilder to avoid re-initialization on
  // subsequent builds. This enables hot reloading. See
  // https://github.com/openfoodfacts/smooth-app/issues/473
  late Future<void> _initFuture;

  @override
  void initState() {
    super.initState();
    _initFuture = _init2();
  }

  Future<bool> _init2() async {
//...
  }

  @override
  Widget build(BuildContext context) {
    return FutureBuilder<void>(
      future: _initFuture,
...
```

Reply:
@monsieurtanuki Done!